python/ray/tune/integration/torch.py

```diff
@@ -51,19 +52,59 @@ class _TorchTrainable(tune.Trainable):
     _function = None
     _num_workers = None
     _use_gpu = None
+    _num_gpus_per_worker = None
     _num_cpus_per_worker = None
+    _num_workers_per_host = None
+    _placement_group = None
+    _timeout_s = None
 
     __slots__ = ["workers", "_finished"]
 
+    @property
+    def worker_gpus(self) -> int:
+        if not self._use_gpu:
+            return 0
+        return self._num_gpus_per_worker
+
+    @property
+    def worker_cpus(self) -> float:
+        return self._num_cpus_per_worker
+
+    @property
+    def should_colocate(self) -> bool:
+        return bool(self._num_workers_per_host)
+
+    @property
+    def num_hosts(self) -> Optional[int]:
+        if self.should_colocate:
+            return int(self._num_workers / self._num_workers_per_host)
+
     @classmethod
-    def default_process_group_parameters(self) -> Dict:
+    def default_process_group_parameters(cls) -> Dict:
         return dict(timeout=timedelta(NCCL_TIMEOUT_S), backend="gloo")
 
-    @classmethod
-    def get_remote_worker_options(self) -> Dict[str, int]:
-        num_gpus = 1 if self._use_gpu else 0
-        num_cpus = int(self._num_cpus_per_worker or 1)
-        return dict(num_cpus=num_cpus, num_gpus=num_gpus)
+    def get_remote_worker_options(self) -> Dict[str, Any]:
+        options = dict(
+            num_cpus=self.worker_cpus, num_gpus=self._num_gpus_per_worker)
+        if self.should_colocate:
+            cpus_per_node = self.worker_cpus * self._num_workers_per_host
+            gpus_per_node = \
+                self._num_gpus_per_worker * self._num_workers_per_host
+            bundle = {}
+            if cpus_per_node:
+                bundle["CPU"] = cpus_per_node
+            if gpus_per_node:
+                bundle["GPU"] = gpus_per_node
+
+            all_bundles = [bundle] * self.num_hosts
+            self._placement_group = placement_group(
+                all_bundles, strategy="STRICT_SPREAD")
+            logger.info("Waiting for placement_group to start.")
+            ray.get(self._placement_group.ready(), timeout=self._timeout_s)
+            logger.info("Placement_group started.")
+            options["placement_group"] = self._placement_group
```

Comment:
this is a repeat of the previous integration; should we move this into `tune.util`?